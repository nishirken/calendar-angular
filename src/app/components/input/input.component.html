<div class="host">
    <wired-input class="input" [value]="value" [placeholder]="placeholder" [type]="type" (input)="onInput.emit($event)" [attr.data-test-id]="testId"></wired-input>
    <div *ngIf="error" class="error" [attr.data-test-id]="errorTestId">{{ error }}</div>
</div>