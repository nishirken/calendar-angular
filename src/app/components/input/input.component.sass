.host
    position: relative

.input
    box-sizing: border-box
    width: 100%

.error
    position: absolute
    bottom: -16px
    color: red
    font-size: 14px
    padding: 0 4px
    box-sizing: border-box