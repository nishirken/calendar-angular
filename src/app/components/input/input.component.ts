import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
  selector: 'app-input',
  templateUrl: './input.component.html',
  styleUrls: ['./input.component.sass'],
})
export class InputComponent {
  @Input() value = '';
  @Input() placeholder = '';
  @Input() error = '';
  @Input() type = '';
  @Input() testId?: string;
  @Input() errorTestId?: string;

  @Output() onInput = new EventEmitter<Event>();
}
