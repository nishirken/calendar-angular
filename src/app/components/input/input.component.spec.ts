import { CUSTOM_ELEMENTS_SCHEMA, EventEmitter } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { InputComponent } from './input.component';

describe('InputComponent', () => {
  let component: InputComponent;
  let fixture: ComponentFixture<InputComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [InputComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
    }).compileComponents();

    fixture = TestBed.createComponent(InputComponent);
    component = fixture.componentInstance;
    component.value = 'Value';
    component.placeholder = 'Placeholder';
    component.type = 'email';
    fixture.detectChanges();
  });

  it('Renders', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('Renders with an error', () => {
    component.error = 'Error';
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
