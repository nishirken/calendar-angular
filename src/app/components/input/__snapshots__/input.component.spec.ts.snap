// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponent Renders 1`] = `
<app-input
  __ngContext__="0"
  error=""
  onInput={[Function EventEmitter_]}
  placeholder={[Function String]}
  type={[Function String]}
  value={[Function String]}
>
  <div
    class="host"
  >
    <wired-input
      class="input"
    />
  </div>
</app-input>
`;

exports[`InputComponent Renders with an error 1`] = `
<app-input
  __ngContext__={[Function Number]}
  error={[Function String]}
  onInput={[Function EventEmitter_]}
  placeholder={[Function String]}
  type={[Function String]}
  value={[Function String]}
>
  <div
    class="host"
  >
    <wired-input
      class="input"
    />
    <div
      class="error"
    >
      Error
    </div>
  </div>
</app-input>
`;
